### PR TITLE
SQL: Non-existing tags in JOIN query result #3945

### PR DESCRIPTION
**Pull request by alexey-kravtsov**

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+import com.axibase.tsd.api.method.series.SeriesMethod;
+import com.axibase.tsd.api.method.sql.SqlTest;
+import com.axibase.tsd.api.model.series.Series;
+import com.axibase.tsd.api.util.Mocks;
+import com.axibase.tsd.api.util.Registry;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
```
Comment (VeselovAlex) [inline src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java]: Remove unnessessary line break

File: src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.axibase.tsd.api.util.TestUtil.TestNames.entity;
+import static com.axibase.tsd.api.util.TestUtil.TestNames.metric;
+
+
```
Comment (VeselovAlex) [inline src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java]: Also here

File: src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        Registry.Entity.register(TEST_ENTITY_NAME);
+        Registry.Metric.register(TEST_METRIC1_NAME);
+        Registry.Metric.register(TEST_METRIC2_NAME);
+
+        String[] allTags = {"tag1", "tag2"};
+        String[] allMetrics = {TEST_METRIC1_NAME, TEST_METRIC2_NAME};
+        List<Series> seriesList = new ArrayList<>();
+
+        for (String tagName : allTags) {
+
```
Comment (VeselovAlex) [inline src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java]: And here

File: src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    /**
+     * #3945
+     */
+    @Test
+    public void testJoinUsingEntityWithTags() {
+        String sqlQuery = String.format(
+                "SELECT t1.tags, t2.tags " +
+                        "FROM '%1$s' t1 JOIN USING ENTITY'%2$s' t2 ",
```
Comment (VeselovAlex) [inline src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java]: Align

File: src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.axibase.tsd.api.util.TestUtil.TestNames.entity;
+import static com.axibase.tsd.api.util.TestUtil.TestNames.metric;
+
+
+public class SqlOuterJoinWithTagsTest extends SqlTest {
+
```
Comment (VeselovAlex) [inline src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java]: Maybe here

File: src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java (hunk 2 above)
Comment (alexey-kravtsov) [inline src/test/java/com/axibase/tsd/api/method/sql/clause/join/SqlOuterJoinWithTagsTest.java]: I think it isn't significant how much (one or two) line breaks between "import" and "class" declarations for code readability (see [EntityTest](https://github.com/axibase/atsd-api-test/blob/master/src/test/java/com/axibase/tsd/api/method/entity/EntityTest.java) or [EntityDeleteTest](https://github.com/axibase/atsd-api-test/blob/master/src/test/java/com/axibase/tsd/api/method/entity/EntityDeleteTest.java) classes)

**Pull request merged**